cajita/unit_test/tstSplineEvaluation2d.hpp

```diff
@@ -0,0 +1,448 @@
+                    point_set.gradient( p, i, j, Dim::I ) =
+                        sd.g[Dim::I][i] * sd.w[Dim::J][j];
+
+                    point_set.gradient( p, i, j, Dim::J ) =
+                        sd.w[Dim::I][i] * sd.g[Dim::J][j];
+                }
+
+            // Evaluate the spline distances at the entities in the stencil.
+            for ( int d = 0; d < 2; ++d )
+                for ( int n = 0; n < ns; ++n )
+                    point_set.distance( p, n, d ) = sd.d[d][n];
+        } );
+}
+
+//---------------------------------------------------------------------------//
+template <class DataTags, class PointCoordinates, class EntityType,
+          int SplineOrder>
+PointSet<typename PointCoordinates::value_type, EntityType, SplineOrder,
+         typename PointCoordinates::device_type, DataTags>
+createPointSet(
+    const PointCoordinates& points, const std::size_t num_point,
+    const std::size_t num_alloc,
+    const LocalGrid<UniformMesh<typename PointCoordinates::value_type, 2>>&
+        local_grid,
+    EntityType, Spline<SplineOrder> )
+{
+    using scalar_type = typename PointCoordinates::value_type;
+
+    using device_type = typename PointCoordinates::device_type;
+
+    using set_type =
+        PointSet<scalar_type, EntityType, SplineOrder, device_type, DataTags>;
+
+    set_type point_set;
+
+    static constexpr int ns = set_type::ns;
+
+    if ( num_point > num_alloc )
+        throw std::logic_error(
+            "Attempted to create point set with more points than allocation" );
+    point_set.num_point = num_point;
+    point_set.num_alloc = num_alloc;
+
+    point_set.cell_size = Kokkos::View<scalar_type* [2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::cell_size" ),
+        num_alloc );
+
+    point_set.logical_coords = Kokkos::View<scalar_type* [2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::logical_coords" ),
+        num_alloc );
+
+    point_set.stencil = Kokkos::View<int* [ns][2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::stencil" ),
+        num_alloc );
+
+    point_set.value = Kokkos::View<scalar_type* [ns][2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::value" ),
+        num_alloc );
+
+    point_set.gradient = Kokkos::View<scalar_type* [ns][ns][2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::gradients" ),
+        num_alloc );
+
+    point_set.distance = Kokkos::View<scalar_type* [ns][2], device_type>(
+        Kokkos::ViewAllocateWithoutInitializing( "PointSet::distance" ),
+        num_alloc );
+
+    auto local_mesh = createLocalMesh<Kokkos::HostSpace>( local_grid );
+
+    int idx_low[2] = { 0, 0 };
+    int idx_low_p1[2] = { 1, 1 };
+    double lowx[2];
+    double lowxp1[2];
+    local_mesh.coordinates( Node(), idx_low, lowx );
+    local_mesh.coordinates( Node(), idx_low_p1, lowxp1 );
+    point_set.dx = lowxp1[0] - lowx[0];
```

Comment:
Could this just use `measure`?

Reply:
This was a way to do this in a dimensionally agnostic way. It would need to be for edges in 3d and faces in 2d. This way I know I always get the cell size. I guess we could also add a special function to the uniform mesh variation of local mesh which returns the cell size directly.